hand-written.md

```diff
@@ -1,46 +1,49 @@
 We are happy to announce the availability of Scala 2.12.0!
 
 The Scala 2.12 compiler has been completely overhauled to make use of the new VM features available in Java 8!
- - A trait is compiled to an interface with default methods for improved binary compatibility.
- - We've unified the treatment of Scala's built-in function types and Single Abstract Method types (for overload resolution, type inference and code generation).
- - Specifically, code generation for function literals uses `invokedynamic` to avoid emitting a classfile at compile time. Many of these features were developed in fruitful collaboration with the Dotty team.
 
-Lukas has delivered a powerful new optimizer with this release, built on earlier work by Miguel Garcia. Many more  (effectively) final methods, including those defined in objects and traits, are now inlined. As well, closure allocations, dead code and [box/unbox pairs](https://github.com/scala/scala/pull/4858) are eliminated more effectively.
+  - A trait is compiled to an interface with default methods for improved binary compatibility.
+  - Lambda syntax can now be used to create instances of types with a single abstract method (SAM types). Scala code can seamlessly use libraries with higher-order functions written in Java 8, and vice versa.
```

Comment:
Mhhh, I guess it only doesn't work when the functions are defined as vals instead of defined inline.